### PR TITLE
To reopen review request

### PR DESCRIPTION
**Pull request by liewteh**

### CODE REVIEW

**Review by scarabeo7 — COMMENTED** [review]
Level 100 is good. Level 200 is also. Level 300, good. Level 400 is good however I noticed it's not in alphabetical order. 
Level 500 is good as page loads all shows on load. Though I noticed that episode dropdown is visible on page load. You probably want to hide that and display it when a show is selected. 

Overall, good work  and I like how the codes and written in blocks.

**Review by nirmeet-baweja — COMMENTED** [review]
# Level   100

1. All episodes must be shown   - Good
2. For each episode, AT LEAST   following must be displayed:
1. the episode's name - Good
2. the season number - Good
3. the episode number - Good
4. the episode's medium-sized   image - Good
5. the episode's summary text   - Good
3. combined season number and episode number into an episode code - Good
1. Each part should be zero-padded to two digits. - Good
2. Example: S02E07 would be the code for the 7th episode of the 2nd season. S2E7 would be incorrect. - Good
4. Your page should state somewhere that the data has (originally) come from TVMaze.com, and link back   to that site (or the specific episode on that site). See   tvmaze.com/api#licensing. - Good
 
# Level 200
 
2. Add a "live"   search input:
1. Only episodes whose summary   OR name contains the search   term should be displayed - Good
2. The search should be case-insensitive - Good
3. The display should update immediately after each keystroke   changes the input. - Good
4. Display how many episodes   match the current search - Good
5. If the search box is   cleared, all episodes should   be shown. - Good
 
 
# Level 300
 
Add a select input which allows you to jump quickly to an episode:
 
1. The select input should   list all episodes in the format: "S01E01 - Winter is Coming"  -Good
2. When the user makes a   selection, they should be taken directly to that episode in the list - Good
3. Bonus: if you prefer, when   the select is used, ONLY show the selected episode. If you do this, be sure   to provide a way for the user to see all episodes again. - Good
 
# Level 350
When your page loads, it must load the   episodes (for the SAME show) from TVMaze API -Good
 
Your search and episode selector must   continue to work - Good
 
Your page MUST NOT re-fetch the episodes   every time the user types a character into your search field
I just noticed that you never declare the variable allEpisodes. So JavaScript will create is as global var. Maybe you should think about creating this as a global variable just like allShows and then you will directly have access to it in other functions as well.
 
# Level 400
Add a select input which allows you to   choose which show you are interested in
Good
 
When a show is selected, your app should   display the episodes for that show
Good
 
Ensure that your search and episode   selector controls still work correctly when you switch shows.
Good
 
This show select must list shows in   alphabetical order, case-insensitive.
Your shows seem to be sorted by rating rather than in alphabetical order. Was it what you planned to do in first place?
 
# Level 500
When your app starts, present a listing   of all shows
Good
 
For each show, you must display at least   name, image, summary, genres, status, rating, and runtime.
Good
 
When a show name is clicked, your app   should fetch and present episodes from that show
Good
 
hide the "shows listing" view
Good 
 
Add a navigation link to enable the user   to return to the "shows listing"
Good.
 
Provide a free-text show search through   show names, genres, and summary texts
Good
 
Ensure that your episode search and   episode selector controls still work correctly when you switch from shows   listing to episodes listing and back
Good, although I noticed that if we select a show and then click on the show selector drop down and select the option "Select Show", it takes us to the Home page with all shows displayed. But the episode selector drop down does not get reset and still shows the episodes of previously displayed show.
 
# Overall comments

Very good work. The code is well broken down into small functions. Maybe in future versions you could make your show blocks a bit more wider, so that the show description could fit one screen and the user does not have to scroll to see it.

**Review by JustinWingChungHui — COMMENTED** [review]
## Level 100

- All episodes must be shown

    Good
- For each episode, AT LEAST following must be displayed:

   - the episode's name - Good
   - the season number - Good
   - the episode number - Good
   - the episode's medium-sized image - Good
   - the episode's summary text - Good


- You should combine season number and episode number into an episode code:
    Good

- Your page should state somewhere that the data has (originally) come from TVMaze.com, and link back to that site (or the specific episode on that site). See tvmaze.com/api#licensing.

    Good


## Level 200

- Add a "live" search input:

    - Only episodes whose summary OR name contains the search term should be displayed - Good
     - The search should be case-insensitive - Good
     - The display should update immediately after each keystroke changes the input. - Good
     - Display how many episodes match the current search - Good
     - If the search box is cleared, all episodes should be shown. - Good

## Level 300
 - Add a select input which allows you to jump quickly to an episode:

     - The select input should list all episodes in the format: "S01E01 - Winter is Coming" - Good
     - When the user makes a selection, they should be taken directly to that episode in the list - Good
     - Bonus: if you prefer, when the select is used, ONLY show the selected episode. If you do this, be sure to provide a way for the user to see all episodes again. - Good

## Level 350

 - When your page loads, it must load the episodes (for the SAME show) from TVMaze API, using fetch, NOT from the provided getAllEpisodes function. (See below for the API "endpoint" (URL) to fetch.) - Good
 - Your search and episode selector must continue to work as specified in level 300. --good
 - Your page MUST NOT re-fetch the episodes every time the user types a character into your search field! - Good

## Level 400
 - Add a select input which allows you to choose which show you are interested in

    - When a show is selected, your app should display the episodes for that show as per the earlier levels of this challenge, except that it should first fetch the episode list from the API - **when you navigate back and forth your code attaches more & more events to the show select, so it fetches multiple times**
    
     - Ensure that your search and episode selector controls still work correctly when you switch shows. - Good
    - This show select must list shows in alphabetical order, case-insensitive. **Not in alphabetical order**


## Level 500
 - When your app starts, present a listing of all shows ("shows listing") - Good
    - For each show, you must display at least name, image, summary, genres, status, rating, and runtime. - Good

 - When a show name is clicked, your app should:

    - fetch and present episodes from that show (enabling episode search and selection as before) - Good
    - hide the "shows listing" view. -Good

- Add a navigation link to enable the user to return to the "shows listing"

    - When this is clicked, the episodes listing should be hidden - Good
    

- Provide a free-text show search through show names, genres, and summary texts. - Good
- Ensure that your episode search and episode selector controls still work correctly when you switch from shows listing to episodes listing and back. - Good 

## Level 999
- Not attempted

## Overall comments
- Works
- Code is reasonably well commented
- Good use of functions
- Well named functions and variables


- Basically formatted
- Some of your functions are quite long, consider breaking them up into smaller ones
- You are attaching events more than once


Completeness 7/10
Code Style 4/5
Clarity 4/5
